### PR TITLE
[adc_ctrl] Minor fixes to adc_ctrl

### PR DESCRIPTION
**Pull request by tjaychen**
- address #10045 by moving "done" 1 cycle back to match captured values.
- update interrupt value on match instead of adc_ctrl_done to ensure we
  only capture interrupt values when there is a match.

### CODE REVIEW

**Review by NigelScales — APPROVED** [review]
I've run with your changed files and the timing looks good now.
The smoke test passes for me.

**N.B. Please don't merge my commit as it is likely out of date now.**
 
image

**Review by rasmus-madsen — APPROVED** [review]
@tjaychen LGTM
let me know if you want me to merge this

**Comment by tjaychen** [comment]
re-based to remove dv commit. I will merge once everything passes.

**Pull request merged**
